openff/arsenic/stats.py

```diff
@@ -209,6 +214,9 @@ def mle(
     for (a, b) in graph.edges:
         i = node_name_to_index[a]
         j = node_name_to_index[b]
+        if i == j:
+            # The MLE solver will fail if we include self-edges, so we need to omit these
+            continue
```

Comment:
Nice catch! This looks good. Just thinking that the loops in lines 193 and 214 could be merged (should increase performance). But that would also require that the loop in line 210 to happen before. It's probably a small increase in performance but thought it should be mentioned. Non-blocking.